src/jvmMain/kotlin/io/beatmaps/util/versions.kt

```diff
@@ -125,16 +129,23 @@ fun publishVersion(mapId: Int, hash: String, alert: Boolean?, rb: RabbitMQInstan
 }
 
 fun pushAlerts(map: BeatmapDao, rb: RabbitMQInstance?) {
+    val allAuthors = (map.collaborators.values + map.uploader).reversed()
+    val authorIds = allAuthors.map { it.id }
+
     val recipients = Follows.selectAll().where {
-        Follows.userId eq map.uploaderId and Follows.upload and Follows.following
+        (Follows.userId inList authorIds) and (Follows.followerId notInList authorIds) and (((Follows.userId eq map.uploaderId) and Follows.upload) or ((Follows.userId neq map.uploaderId) and Follows.collab)) and Follows.following
     }.map { row ->
         row[Follows.followerId].value
     }
```

Comment:
If a user follows more than one author this query returns them twice causing a duplicate key exception when inserting alerts below.